mesheryctl/internal/cli/root/system/start.go

```diff
@@ -181,22 +181,23 @@ func start() error {
 		return errors.Wrap(err, "error processing config")
 	}
 
-	// TODO: Centralize docker-compose.yaml file pull for this and reset
 	// get the platform, channel and the version of the current context
-	currPlatform := mctlCfg.GetContextContent().Platform
-	RequestedAdapters := mctlCfg.GetContextContent().Adapters // Requested Adapters / Services
-	// TODO: Context support for system start
-	// currChannel := mctlCfg.Contexts[currentContext].Channel
-	// currVersion := mctlCfg.Contexts[currentContext].Version
-	// fileURL := ""
+	// if a temp context is set using the -c flag, use it as the current context
+	currCtxName, currCtx, err := utils.GetCurrentContext(tempContext)
+	if err != nil {
+		return err
+	}
+	currPlatform := mctlCfg.Contexts[currCtxName].Platform
+	RequestedAdapters := mctlCfg.Contexts[currCtxName].Adapters // Requested Adapters / Services
 
 	// Deploy to platform specified in the config.yaml
 	switch currPlatform {
 	case "docker":
 
 		if _, err := os.Stat(utils.MesheryFolder); os.IsNotExist(err) {
-			if err := utils.DownloadFile(utils.DockerComposeFile, fileURL); err != nil {
-				return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to download %s file from %s", utils.DockerComposeFile, fileURL)))
+			// download the docker-compose.yaml file corresponding to the current version
+			if _, err := utils.DownloadDockerComposeFile(currCtx, true); err != nil {
+				return errors.Wrapf(err, utils.SystemError(fmt.Sprintf("failed to download %s file", utils.DockerComposeFile)))
 			}
 		}
```

Comment:
@navendu-pottekkat ` utils.DownloadDockerComposeFile` already handles checking if file exists, that's why we had added `force` argument.
The above code currently skips downloading the file if the folder exists, thus skipping our forceful download. (It also skips the case when `.meshery` folder exists but no `meshery.yaml` file in that folder.
Should I fix this change when working on log and update command child issues?

Reply:
@aneeshnema The force argument is set to true right? So it should download the files, shouldn't it?

Reply:
@navendu-pottekkat  yes, it will.
I am talking about this,
`		if _, err := os.Stat(utils.MesheryFolder); os.IsNotExist(err) {`
This would skip the inside code block altogether if the `.meshery` folder exists.

Reply:
Yep, you could do that. This was done since the meshery.yaml file was recreated in the system start command. Changing that should not break anything. But lets make sure that is the case. Please add it in the PR.